### PR TITLE
[Java] Adding support for ACL and DML providers

### PR DESCRIPTION
**Pull request by Craigacp**
**Description**:
Adds the pass through methods to session options which allow choosing the DirectML and ARM Compute Library providers from the Java API.

### CODE REVIEW

**Comment by snnn** [comment]
Our CI is broken now. I'll schedule the build when it is fixed.

File: java/src/main/java/ai/onnxruntime/OrtSession.java [hunk 1]
```diff
@@ -548,6 +548,26 @@ public void addNuphar(boolean allowUnalignedBuffers, String settings) throws Ort
       addNuphar(OnnxRuntime.ortApiHandle, nativeHandle, allowUnalignedBuffers ? 1 : 0, settings);
     }
 
+    /**
+     * Adds DirectML as an execution backend.
+     *
+     * @param deviceNum The id of the DirectML device.
+     * @throws OrtException If there was an error in native code.
+     */
+    public void addDirectML(int deviceNum) throws OrtException {
```
Comment (pranavsharma) [inline java/src/main/java/ai/onnxruntime/OrtSession.java:557]: nit: I would call this device id to be consistent with the C API. Same for documentation above.

Comment (Craigacp) [reply java/src/main/java/ai/onnxruntime/OrtSession.java]: Done.

**Pull request merged**